app/src/main/java/de/Maxr1998/xposed/maxlock/ui/lockscreen/LockView.java

```diff
@@ -83,12 +90,20 @@ public LockView(Context context, String packageName, String activityName) {
         mActivityName = activityName;
 
         mLockingType = getPreferencesKeysPerApp(getContext()).getString(mPackageName, getPrefs().getString(Common.LOCKING_TYPE, ""));
-
-        if (getPreferencesKeysPerApp(getContext()).contains(mPackageName)) {
-            mPassword = getPreferencesKeysPerApp(getContext()).getString(mPackageName + Common.APP_KEY_PREFERENCE, null);
-        } else
-            mPassword = getPreferencesKeys(getContext()).getString(Common.KEY_PREFERENCE, "");
-
+        Log.e(LockView.class.getName(), mLockingType);
+        if (mLockingType.equals(Common.PREF_VALUE_GESTURE)) {
+            if (getPreferencesKeysPerApp(getContext()).contains(mPackageName))
+                mPassword = mPackageName + Common.APP_KEY_GESTURE;
+            else
+                mPassword = Common.MASTER_GESTURE_NAME;
+            }
+        else {
+            if (getPreferencesKeysPerApp(getContext()).contains(mPackageName))
+                mPassword = getPreferencesKeysPerApp(getContext()).getString(mPackageName + Common.APP_KEY_PREFERENCE, null);
+            else
+                mPassword = getPreferencesKeys(getContext()).getString(Common.KEY_PREFERENCE, "");
+        }
+        Log.e(LockView.class.getName(), mPassword);
```

Comment:
Apologies, that was for debugging. Right now I've exams, so I'll remove it
after ~13 hours (two exams today).

On Mon 21 Mar, 2016, 02:52 Max Rumpf, <notifications@github.com> wrote:

> In
> app/src/main/java/de/Maxr1998/xposed/maxlock/ui/lockscreen/LockView.java
> <https://github.com/Maxr1998/MaxLock/pull/81#discussion_r56772396>:
>
> > -            mPassword = getPreferencesKeys(getContext()).getString(Common.KEY_PREFERENCE, "");
> > -
> > +        Log.e(LockView.class.getName(), mLockingType);
> > +        if (mLockingType.equals(Common.PREF_VALUE_GESTURE)) {
> > +            if (getPreferencesKeysPerApp(getContext()).contains(mPackageName))
> > +                mPassword = mPackageName + Common.APP_KEY_GESTURE;
> > +            else
> > +                mPassword = Common.MASTER_GESTURE_NAME;
> > +            }
> > +        else {
> > +            if (getPreferencesKeysPerApp(getContext()).contains(mPackageName))
> > +                mPassword = getPreferencesKeysPerApp(getContext()).getString(mPackageName + Common.APP_KEY_PREFERENCE, null);
> > +            else
> > +                mPassword = getPreferencesKeys(getContext()).getString(Common.KEY_PREFERENCE, "");
> > +        }
> > +        Log.e(LockView.class.getName(), mPassword);
>
> Could you remove logging th password nd locking type, please?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Maxr1998/MaxLock/pull/81/files/37e91071dc7d501d6e11d4ef43350553ef4d4ef8#r56772396>
>